From ba1bbccc811f89076a1f19da05c7db374df750b9 Mon Sep 17 00:00:00 2001
From: gavanderhoorn <g.a.vanderhoorn@tudelft.nl>
Date: Fri, 24 Sep 2021 18:18:51 +0200
Subject: [PATCH 1/7] Help debug

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c32a3d..b694cbd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,8 +50,10 @@ if(NOT WIN32)
 endif()
 
 if(WIN32)
+  message(STATUS "Win32 time.c")
   set(time_impl_c src/time_win32.c)
 else()
+  message(STATUS "Unix time.c")
   set(time_impl_c src/time_unix.c)
 endif()
 
-- 
2.43.0

