From df6c9fd7f85abea288f49ec79452ba5cf93d0c23 Mon Sep 17 00:00:00 2001
From: gavanderhoorn <g.a.vanderhoorn@tudelft.nl>
Date: Fri, 24 Sep 2021 18:20:27 +0200
Subject: [PATCH 3/7] No support for deprecation msg

---
 include/rcutils/macros.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/rcutils/macros.h b/include/rcutils/macros.h
index 2e42163..c9af7da 100644
--- a/include/rcutils/macros.h
+++ b/include/rcutils/macros.h
@@ -118,8 +118,12 @@ extern "C"
 
 /// Macro to declare deprecation in the platform appropriate manner with a message.
 #ifndef _WIN32
+#ifndef YASKAWA_MOTOMAN_MOTOPLUS1_GCC43
 # define RCUTILS_DEPRECATED_WITH_MSG(msg) __attribute__((deprecated(msg)))
 #else
+# define RCUTILS_DEPRECATED_WITH_MSG(msg) __attribute__((deprecated))
+#endif // YASKAWA_MOTOMAN_MOTOPLUS1_GCC43
+#else
 # define RCUTILS_DEPRECATED_WITH_MSG(msg) __declspec(deprecated(msg))
 #endif
 
-- 
2.43.0

