From c19015f699087026c1ea80814802cfcbbbd1bb4b Mon Sep 17 00:00:00 2001
From: gavanderhoorn <g.a.vanderhoorn@tudelft.nl>
Date: Tue, 12 Oct 2021 19:07:17 +0200
Subject: [PATCH 5/7] Platform has DIR, don't re-typedef

---
 src/filesystem.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/filesystem.c b/src/filesystem.c
index be6f764..5fdcc66 100644
--- a/src/filesystem.c
+++ b/src/filesystem.c
@@ -55,7 +55,9 @@
 #endif  // _WIN32
 
 #ifdef RCUTILS_NO_FILESYSTEM
+#ifndef YASKAWA_MOTOMAN_MOTOPLUS1 // TODO: we're using this as a platform ifdef
 typedef int DIR;
+#endif  // _YASKAWA_MOTOMAN_MOTOPLUS1
 #endif  // _RCUTILS_NO_FILESYSTEM
 
 typedef struct rcutils_dir_iter_state_s
-- 
2.43.0

