From cfb23f620bafa498a632e9634bd1213ed58cb9a5 Mon Sep 17 00:00:00 2001
From: jimmy-mcelwain <jimmy.mcelwain@motoman.com>
Date: Thu, 11 Jun 2026 13:56:55 -0400
Subject: [PATCH 7/7] disable fork/exec/wait

---
 src/process.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/process.c b/src/process.c
index 509ef65..a696ccc 100644
--- a/src/process.c
+++ b/src/process.c
@@ -36,7 +36,9 @@ extern "C"
 #pragma warning(pop)
 #else
 #include <libgen.h>
+#ifndef YASKAWA_MOTOMAN_MOTOPLUS1  // Platform has no fork/exec/wait support
 #include <sys/wait.h>
+#endif  // YASKAWA_MOTOMAN_MOTOPLUS1
 #include <unistd.h>
 #endif
 
@@ -290,6 +292,11 @@ rcutils_start_process(
   process->pid = process_info.dwProcessId;
 
   return process;
+#elif defined(YASKAWA_MOTOMAN_MOTOPLUS1)  // Platform has no fork/exec/wait support
+  (void)args;
+  RCUTILS_SET_ERROR_MSG("subprocess support is not available on this platform");
+  rcutils_process_close(process);
+  return NULL;
 #else
   char **argv = allocator->zero_allocate(args->size + 1, sizeof(*argv), &allocator->state);
   if (NULL == argv) {
@@ -356,6 +363,10 @@ rcutils_process_wait(const rcutils_process_t * process, int * exit_code)
     }
     *exit_code = status;
   }
+#elif defined(YASKAWA_MOTOMAN_MOTOPLUS1)  // Platform has no fork/exec/wait support
+  (void)exit_code;
+  RCUTILS_SET_ERROR_MSG("subprocess support is not available on this platform");
+  return RCUTILS_RET_ERROR;
 #else
   int status;
 
-- 
2.43.0

